tests/pos/abstract-inline-val.scala

```diff
@@ -0,0 +1,11 @@
+trait C:
+  inline def x: Int
+  inline val y: Int
+
+class C1 extends C:
+  inline val x = 1
+  inline val y = 2
```

Comment:
A side note: in `inline val`, we require the rhs to be a literal type, which is not really related to `inline`. Meanwhile, const-folding can be achieved by annotating the constant type explicitly.

Reply:
Actually, inline val requires a literal type result type and a pure body. This is a consequence of being marked as inline, as these are the only values that can be soundly inlined. We use constant folding to inline it because thee type already provides this functionality.

Reply:
It's more about semantics than soundness. There is an inconsistency between _inline vals_ and _inline parameters_.